src/tootsee/ports/mastodon_stream.cr

```diff
@@ -0,0 +1,34 @@
+module Tootsee
+  module Ports
+    # Incoming port from the Mastodon streaming client.
+    abstract class MastodonStream
+      # This does not return. It calls the block whenever the bot receives a
+      # push notification from Mastodon.
+      abstract def listen(&block : Mastodon::Entities::Notification -> Void)
```

Comment:
Since you used `Mastodon::Entities::Notification` a couple of times in the class, it might make sense to define this in the  class? `Notification = 
Mastodon::Entities::Notification`

Reply:
Hmm I'd rather preserve clarity of where things are coming from